### ISSUE TITLE
Int64.sub inconsistent result on JS target

### ISSUE THREAD

**Issue by lbergman** [HaxeFoundation/haxe#7485]
http://try-haxe.mrcdk.com/#88D6b

```
var a = haxe.Int64.parseString('2147483647');
var b = haxe.Int64.parseString('9223372036854775807');
trace(haxe.Int64.toStr(haxe.Int64.sub(a, b)));
```

Returns the following:
-9223372030412324864 <- JS
-9223372034707292160 <- Flash and Neko

Tested on 3.4.4 and latest nightly on mrcdk

**Comment by Simn** [comment]
That looks like something isn't clamped right somewhere.

**Comment by jcward** [comment]
FYI, in hex:
```
0x7fffffff - 0x7fffffffffffffff =

     JS: -7ffffffd80000000
Correct: -7fffffff80000000
```
The largest positive 32-bit integer minus the largest positive 64-bit integer.

The subtraction (seemingly correctly) produces this `Int64`, which prints wrong on js:

```
    var z = haxe.Int64.make(-2147483648, -2147483648);
    trace(haxe.Int64.toStr(z));

     js: -9223372030412324864
correct: -9223372034707292160
```

**Comment by jcward** [comment]
Ah, something's wrong in the negate behavior of this number on JS (invoked via `toStr()` -> `divMod()` -> `var modulus = -dividend`):

```
    var z = haxe.Int64.make(-2147483648, -2147483648);
    trace(z.low);
    trace((-z).low); // JS: 2147483648   Others: -2147483648
```

Negate invokes [-x.low](https://github.com/HaxeFoundation/haxe/blob/development/std/haxe/Int64.hx#L218), which boils down to an `Int32.negate` inconsistency in JS:

```
    var a:Int = -2147483648;
    trace(a);
    trace(-a);   // JS: 2147483648   Others: -2147483648
```

**Comment by jcward** [comment]
Please comment on whether fix belongs in `Int64` or `Int32` in the PR, #7487

**Comment by Simn** [comment]
This has been addressed in #7491.
